homeassistant/components/alexa/capabilities.py

```diff
@@ -922,108 +909,100 @@ def properties_proactively_reported(self):
 
     def properties_retrievable(self):
         """Return True if properties can be retrieved."""
+        return True
 
     def get_property(self, name):
         """Read and return a property."""
         if name != "mode":
             raise UnsupportedProperty(name)
 
+        # Fan Direction
         if self.instance == f"{fan.DOMAIN}.{fan.ATTR_DIRECTION}":
-            return self.entity.attributes.get(fan.ATTR_DIRECTION)
+            mode = self.entity.attributes.get(fan.ATTR_DIRECTION, None)
+            if mode in (fan.DIRECTION_FORWARD, fan.DIRECTION_REVERSE, STATE_UNKNOWN):
+                return f"{fan.ATTR_DIRECTION}.{mode}"
 
+        # Cover Position
         if self.instance == f"{cover.DOMAIN}.{cover.ATTR_POSITION}":
-            return self.entity.attributes.get(cover.ATTR_POSITION)
+            # Return state instead of position when using ModeController.
+            mode = self.entity.state
+            if mode in (
+                cover.STATE_OPEN,
+                cover.STATE_OPENING,
+                cover.STATE_CLOSED,
+                cover.STATE_CLOSING,
+                STATE_UNKNOWN,
+            ):
+                return f"{cover.ATTR_POSITION}.{mode}"
 
         return None
 
     def configuration(self):
         """Return configuration with modeResources."""
-        return self.serialize_mode_resources()
+        if isinstance(self._resource, AlexaCapabilityResource):
+            return self._resource.serialize_configuration()
 
     def capability_resources(self):
         """Return capabilityResources object."""
-        capability_resources = []
 
+        # Fan Direction Resource
         if self.instance == f"{fan.DOMAIN}.{fan.ATTR_DIRECTION}":
-            capability_resources = [
-                {"type": Catalog.LABEL_ASSET, "value": Catalog.SETTING_DIRECTION}
-            ]
+            self._resource = AlexaModeResource(
```

Comment:
Isn't this a bit weird in that we now rely inside `configuration()` on the side effect of this method to set `self._resource` ? 

Couldn't we have something like `get_resource` that is used by both this method and `configuration()` ?

Reply:
`capability_resources()` does not rely inside `configuration()`. Both `capability_resources()` and `configuration()` still override the individual static methods in the `AlexaCapability` base class. 

I see where I created confusion; `configuration()` implicitly returns `None` if `isinstance(self._resource, AlexaCapabilityResource)` is `False`. For capability interfaces that do not need to describe the configuration object, `configuration()` should return `None` or evaluate to `False` so it's not included in the `serialize_discovery()` response. 

Quick Reference: Regarding Alexa discovery response while `capabilityResources` object and `configuration` object appear to be related and are used together to describe the `RangeController`, `ModeController` interfaces; Alexa doesn't require any related property value to link them together. All they do is translate custom names for setting names, and setting values for the voice model. Furthermore the `configuration` object is not used for the `ToggleController`, as it just uses "on" and "off" for the voice model setting values.

Reply:
@balloob Resolved conflicts and added explicit return None for configuration().